Develop

BadGreyFox opened the pull request:
Проект 4-го спринта, Ерофеев

BadGreyFox committed:
add pom.xml

BadGreyFox committed:
Проект 4-го спринта, Ерофеев

pomegranate282922 commented on src/test/java/ru/praktikum_services/scooter/FaqAnswersTextTest.java:
```diff
@@ -0,0 +1,67 @@
+                    .clickToQuestionPanel(value.indexOfQuestion)
+                    .getNotHiddenAnswer();
+            assertEquals("Действительный ответ не совпадает с ожидаемым", value.isAnswerEqualExpected, actualAnswer.equals(value.expectedAnswers));
+        }
```
Так проверяет получились не независимыми. Если упадет первая - другие не проверятся. Нужно сами вопросы сделать в параметризации, без циклов 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum_services/scooter/PlacingOrderTest.java:
```diff
@@ -0,0 +1,129 @@
+                .clickOrderConfirmationButton()
+                .checkOrderIsProcessed();
+        assertEquals("Заказ не создан, страница с подтверждением заказа не отобразилась", isOrderProcessedExpected, isOrderProcessedActual);
+
```
Здесь нужно немного изменить логику теста.
 Первый флоу ( верхняя кнопка) - тестируем минимум двумя тестовыми наборами до появления надписи Заказ оформлен.
 Второй флоу ( нижняя кнопка) - тестируем до появления формы ввода параметров заказа.
 Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum_services/scooter/BaseTest.java:
```diff
@@ -0,0 +1,46 @@
+    protected WebDriver driver;
+    @Before
+    public void setUp(){
+        String browser = System.getenv("BROWSER");
```
Отлично что выбор браузера делается извне

pomegranate282922 reviewed (commented):
Хорошая работа! Тесты реализованы с помощью Page Object Model. Применена параметризация ( правда, не совсем так как ожидаем). Здорово  реализован выбор браузера. 
Замечания:
1. Реализовать параметризацию так, чтобы все 8 проверок были независимыми 
2. Поменять логику в тесте на создание заказа

BadGreyFox committed:
Проект 4-го спринта, Ерофеев:
Исправлена логика теста для FAQ:
 1. Параметризованы сами ответы, убран цикл проверки;
 2. В ассерте сравниваем строки, а не булевые значения.
 Исправлена логика теста для Заказа:
 1. Исправлена проверка отображения с подтверждением формы заказа - проверяем появляется ли надпись "Заказ оформлен";
 2. Исправлена проверка для нижней кнопки Заказа - проверяем, появилась ли форма для заполнения пер.данных.

pomegranate282922 commented on src/test/java/ru/praktikum_services/scooter/PlacingOrderTest.java:
```diff
@@ -0,0 +1,70 @@
+    }
+    //Проверка с нижней кнопкой Заказа - появляется ли форма для заполнения данных
+    @Test
+    public void checkingCorrectnessOrderWithBottomButtonPositive() {
```
Этот тест повторится дважды так как лежит в параметризованном классе.
Можно улучшить

pomegranate282922 commented on src/test/java/ru/praktikum_services/scooter/PlacingOrderTest.java:
```diff
@@ -0,0 +1,70 @@
+        boolean isDisplayedFormPanel = new HeaderPage(driver)
+                .clickBottomOrderButton()
+                .isDisplayedFormPanel();
+        assertTrue("Заказ не создан, страница с подтверждением заказа не отобразилась", isDisplayedFormPanel);
```
Текст сообщения лучше поправить - мы здесь и не ожидаем создание заказа, правда? 
Можно улучшить

pomegranate282922 reviewed (approved):
Ок, критичных замечаний нет! Зачет!
